wpimath/src/test/java/edu/wpi/first/math/geometry/QuaternionTest.java

```diff
@@ -47,6 +47,58 @@ void testInit() {
             + q3.getZ() * q3.getZ());
   }
 
+  @Test
+  void testAddition() {
+    var q = new Quaternion(0.1, 0.2, 0.3, 0.4);
+    var p = new Quaternion(0.5, 0.6, 0.7, 0.8);
+
+    var sum = q.plus(p);
+
+    assertEquals(q.getW() + p.getW(), sum.getW());
+    assertEquals(q.getX() + p.getX(), sum.getX());
+    assertEquals(q.getY() + p.getY(), sum.getY());
+    assertEquals(q.getZ() + p.getZ(), sum.getZ());
```

Comment:
Nit: This (and others) could be changed to `assertAll(() -> assertEquals(...), () -> assertEquals(...))` so that if it does fail, all of the values will be displayed (instead of just the X value).

Reply:
fixed